discovery/v1.js

```diff
@@ -0,0 +1,331 @@
+  if (typeof this._options.version_date === 'undefined') {
+    throw new Error('Argument error: version_date was not specified, use DiscoveryV1.VERSION_DATE_2016_07_11');
+  }
+  this._options.qs.version = options.version_date;
+}
+
+util.inherits(DiscoveryV1, BaseService);
+DiscoveryV1.prototype.name = 'discovery';
+DiscoveryV1.prototype.version = 'v1-experimental';
+DiscoveryV1.URL = 'https://gateway.watsonplatform.net/discovery-experimental/api';
+
+
+/**
+ * Initial release
+ * @type {string}
+ */
+DiscoveryV1.VERSION_DATE_2016_07_11 = '2016-11-07';
+
+
+
+/**
+ * Return the list of environments
+ *
+ * @param {Object} params
+ * @param {String} [params.name] search enviroments with the given name
+ */
+DiscoveryV1.prototype.getEnvironments = function(params, callback) {
+  params = params || {};
+
+  var parameters = {
+    options: {
+      url: '/v1/environments',
+      method: 'GET',
+      json: true
+    },
+    defaultOptions: this._options
+  };
+  return requestFactory(parameters, callback);
+};
+
+/**
+ * Get details about an environment
+ *
+ * @param {Object} params
+ * @param {String} params.environment_id
+ */
+DiscoveryV1.prototype.getEnvironment = function(params, callback) {
+  params = params || {};
+
+  var parameters = {
+    options: {
+      url: '/v1/environments/{environment_id}',
+      method: 'GET',
+      path: pick(params, ['environment_id']),
+      json: true
+    },
+    requiredParams: ['environment_id'],
+    defaultOptions: this._options
+  };
+  return requestFactory(parameters, callback);
+};
+
+/**
+ * List all configurations
+ *
+ * @param {Object} params
+ * @param {String} params.environment_id
+ */
+DiscoveryV1.prototype.getConfigurations = function(params, callback) {
+  params = params || {};
+
+  var parameters = {
+    options: {
+      url: '/v1/environments/{environment_id}/configurations',
+      method: 'GET',
+      path: pick(params, ['environment_id']),
+      json: true
+    },
+    requiredParams: ['environment_id'],
+    defaultOptions: this._options
+  };
+  return requestFactory(parameters, callback);
+};
+
+/**
+ * Get details about a configuration
+ *
+ * @param {Object} params
+ * @param {String} params.environment_id
+ * @param {String} params.configuration_id
+ */
+DiscoveryV1.prototype.getConfiguration = function(params, callback) {
+  params = params || {};
+
+  var parameters = {
+    options: {
+      url: '/v1/environments/{environment_id}/configurations/{configuration_id}',
+      method: 'GET',
+      path: pick(params, ['environment_id','configuration_id']),
+      json: true
+    },
+    requiredParams: ['environment_id','configuration_id'],
+    defaultOptions: this._options
+  };
+  return requestFactory(parameters, callback);
+};
+
+/**
+ * Return the list of collections in the given environment
+ *
+ * @param {Object} params
+ * @param {String} params.environment_id
+ * @param {String} [params.name] Find collections with the given name.
```

Comment:
I don't see params.name used anywhere in this method.

(It looks like this is also wrong in https://github.com/watson-developer-cloud/node-sdk/blob/master/discovery/v1-experimental.js#L103 )